src/DataShare/DataShare/ProviderShareSubscriptions/GrantAzDataShareSubscriptionAccess.cs

```diff
@@ -0,0 +1,123 @@
+         SupportsShouldProcess = true),
+     OutputType(typeof(PSDataShareProviderShareSubscription))]
+    public class GrantAzDataShareSubscriptionAccess : AzureDataShareCmdletBase
+    {
+        /// <summary>
+        /// The resource group name of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name of the azure data share account",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure data share account name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure data share name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Share, "ResourceGroupName", "AccountName")]
+        public string ShareName { get; set; }
+
+        /// <summary>
+        /// The share subscription id of the provider share subscription.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The share subscription id of the provider share subscription",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The share subscription id of the provider share subscription",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ResourceIdCompleter(ResourceTypes.ShareSubscription)]
+        [ValidateNotNullOrEmpty]
+        public string ShareSubscriptionId { get; set; }
+
+        /// <summary>
+        /// The resourceId of the azure data share.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource id of the azure data share",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ResourceIdCompleter(ResourceTypes.Share)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
```

Comment:
No AsJob parameter?  Grant is not long-running, but revoke is?

Reply:
No it isnt a long-running job